### PR TITLE
Don't abuse daemonset env condition

### PR DESCRIPTION
**Pull request by toshipp**
Previous code used daemonset env condition in many places, but it did not directly express the meaning.
In this PR, change the condition or remove it to make it more readable.

### CODE REVIEW

File: e2e/suite_test.go [hunk 1]
```diff
@@ -96,13 +110,18 @@ var _ = BeforeSuite(func() {
...
-	if !isDaemonsetLvmdEnvSet() {
-		By("Waiting for kindnet to get ready")
-		// Because kindnet will crash. we need to confirm its readiness twice.
-		Eventually(waitKindnet).Should(Succeed())
-		time.Sleep(5 * time.Second)
-		Eventually(waitKindnet).Should(Succeed())
-	}
+	By("Getting node count")
+	var nodes corev1.NodeList
+	err := getObjects(&nodes, "nodes", "-l=node-role.kubernetes.io/control-plane!=")
```
Comment (cupnes) [inline e2e/suite_test.go:115]: About filtering by label, the right-hand side of `node-role.kubernetes.io/control-plane!=` is not set, is this correct?

I tried the following and it shows both control-plane node and non-control-plane node.

```
kubectl get nodes -l=node-role.kubernetes.io/control-plane!=
```

Comment (toshipp) [reply e2e/suite_test.go:115]: @cupnes 
In my kind environment (v1.27 k8s), topolvm-e2e-control-plane has the following labels.
```
beta.kubernetes.io/arch: amd64
beta.kubernetes.io/os: linux
kubernetes.io/arch: amd64
kubernetes.io/hostname: topolvm-e2e-control-plane
kubernetes.io/os: linux
node-role.kubernetes.io/control-plane: ""
node.kubernetes.io/exclude-from-external-load-balancers: ""
```

The expression expects NOT to match the empty value of the key `node-role.kubernetes.io/control-plane`, so it is intentional and I can get only non-controller nodes only.

 What environment did you try it in?

Comment (toshipp) [reply e2e/suite_test.go:115]: Here is my result.
without the label selector:
```
$ kubectl get node --show-labels
NAME                        STATUS   ROLES           AGE   VERSION   LABELS
topolvm-e2e-control-plane   Ready    control-plane   10m   v1.27.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=topolvm-e2e-control-plane,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node.kubernetes.io/exclude-from-external-load-balancers=
topolvm-e2e-worker          Ready    <none>          10m   v1.27.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=topolvm-e2e-worker,kubernetes.io/os=linux,topology.topolvm.io/node=topolvm-e2e-worker
topolvm-e2e-worker2         Ready    <none>          10m   v1.27.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=topolvm-e2e-worker2,kubernetes.io/os=linux,topology.topolvm.io/node=topolvm-e2e-worker2
topolvm-e2e-worker3         Ready    <none>          10m   v1.27.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=topolvm-e2e-worker3,kubernetes.io/os=linux,topology.topolvm.io/node=topolvm-e2e-worker3
```

with the label selector:
```
$ kubectl get node --show-labels -lnode-role.kubernetes.io/control-plane!=
NAME                  STATUS   ROLES    AGE   VERSION   LABELS
topolvm-e2e-worker    Ready    <none>   10m   v1.27.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=topolvm-e2e-worker,kubernetes.io/os=linux,topology.topolvm.io/node=topolvm-e2e-worker
topolvm-e2e-worker2   Ready    <none>   10m   v1.27.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=topolvm-e2e-worker2,kubernetes.io/os=linux,topology.topolvm.io/node=topolvm-e2e-worker2
topolvm-e2e-worker3   Ready    <none>   10m   v1.27.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=topolvm-e2e-worker3,kubernetes.io/os=linux,topology.topolvm.io/node=topolvm-e2e-worker3
```

Comment (toshipp) [reply e2e/suite_test.go:115]: In another environment, I found nodes have `node-role.kubernetes.io/control-plane=true` label. I don't why the values differ, I'll change the expression to check the key only.

File: e2e/sanity_test.go [hunk 2]
```diff
@@ -35,47 +37,47 @@ func testSanity() {
 			g.Expect(err).ShouldNot(HaveOccurred())
 			g.Expect(ds.Status.NumberAvailable).To(BeEquivalentTo(1))
 		}).Should(Succeed())
-	})
```
Comment (llamerada-jp) [inline e2e/sanity_test.go:38]: It seems that no need to create tc and thinTC in a BeforeEach block. I would like to know if there is a reason.

Comment (toshipp) [reply e2e/sanity_test.go:38]: It is because Address or other fields depend on nonControlPlaneNodeCount which will be initialized in BeforeSuite.

Comment (toshipp) [reply e2e/sanity_test.go:38]: I noted it here: https://github.com/topolvm/topolvm/pull/807/files#diff-df24cc5d6cd0a9d1708368dc0c4119fc7ac5ef7ef7293bc004d64985d736739fR21

Comment (llamerada-jp) [reply e2e/sanity_test.go:38]: Thank you.

**Review by llamerada-jp — APPROVED** [review]

**Review by cupnes — APPROVED** [review]

**Pull request merged**